### PR TITLE
[WIP] [install] allow parallel execution of `postinstall` scripts

### PR DESCRIPTION
**Pull request by alexlamsl**
- fixes spurious exit code 42 from `spawn()`

### What does this PR do?

TODO: suppress log output from `postinstall` scripts instead of spewing all over the console &minus; only display logs from failed scripts to aid debugging.

Making this PR now because during development I have discovered the root cause for the `exitCode=42` bug, so seems like a good point to check this in.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

I wrote automated tests

If Zig files changed:

- [x] I checked the lifetime of memory allocated to verify it's (1) freed and (2) only freed when it should be
- [x] I or my editor ran `zig fmt` on the changed files
- [x] I included a test for the new code, or an existing test covers it
- [ ] JSValue used outside outside of the stack is either wrapped in a JSC.Strong or is JSValueProtect'ed

### CODE REVIEW

File: src/cli/run_command.zig [hunk 1]
```diff
@@ -265,57 +310,73 @@ pub const RunCommand = struct {
...
-                        Output.flush();
-                    }
-
-                    Global.exit(code);
-                }
-            },
-            .Signal => |signal| {
-                Output.prettyErrorln("<r><red>error<r><d>:<r> script <b>\"{s}\"<r> exited with {any}<r>", .{ name, bun.SignalCode.from(signal) });
+    pub fn waitForPackageScript(name: string, pid: std.os.pid_t, is_sync: bool) bool {
+        switch (PosixSpawn.waitpid(pid, if (is_sync) 0 else std.os.W.NOHANG)) {
```
Comment (Jarred-Sumner) [inline src/cli/run_command.zig]: this needs a `while (true)` when `is_sync` incase signals are sent but there's no action necessary

File: src/cli/run_command.zig [hunk 2]
```diff
@@ -225,26 +226,70 @@ pub const RunCommand = struct {
...
+        }
+
+        var attr = try PosixSpawn.Attr.init();
+        defer attr.deinit();
+        try attr.set(@intCast(flags));
+        try attr.resetSignals();
+
+        var actions = try PosixSpawn.Actions.init();
+        defer actions.deinit();
+        try actions.inherit(0);
```
Comment (Jarred-Sumner) [inline src/cli/run_command.zig:248]: I don't think we can ship this with `inherit`. It's going to be too noisy.

Comment (alexlamsl) [reply src/cli/run_command.zig:248]: sounds reasonable &minus; will work on this now

File: src/cli/run_command.zig (hunk 1 above)
Comment (alexlamsl) [reply src/cli/run_command.zig]: done in https://github.com/oven-sh/bun/pull/4475/commits/9bafac63900647655a48bf0be4c2bea8e71fdc2e

**Comment by paperclover** [comment]
was merged to #5077

**Pull request closed without merging**